core/chains/evm/client/rpc_client.go

```diff
@@ -147,7 +191,7 @@ func NewRPCClient(
 	id int32,
 	chainID *big.Int,
 	tier commonclient.NodeTier,
-) *RpcClient {
```

Comment:
Is there reason to start returning interface instead of struct? I'd prefer struct

Reply:
This was done to enable testing to `BatchCallContext` which required a mockRPC to implement different functionality. However, I can try to find a different way to test this, in which case we can return the struct and even remove `ChainClientRPC` type completely.